### DISCUSSION TITLE
Introspection query result to schema SDL

### DISCUSSION THREAD

**Discussion by guidomb** [apollographql/apollo-rs#557] [Q&A] [answered]
Hi, I'm currently working on a project where I need to parse the output of a schema introspection query and then generate the schema in SDL format. Looks like I could export the schema in SDL using https://docs.rs/apollo-encoder/latest/apollo_encoder/ but I wonder if there is an easy way of creating the `Document` instance from the introspection query output without having to parse it manually.

Thanks!

**Comment by abernix** [comment, marked as answer]
I'm not certain if this helps you, but we have a crate called `introspector_gadget` that could be along the lines of what you're trying to do (or help!): https://github.com/apollographql/introspector-gadget.  It uses `apollo_encoder` to print the SDL and accepts the JSON representation of the schema.

**Reply by guidomb** [reply]
Thanks this worked! As a future reference for anyone reading this. Here is an example program:

```rust
use std::{collections::HashMap, eprintln, println};

use introspector_gadget::{
    blocking::GraphQLClient,
    introspect::{self, GraphIntrospectInput},
};

fn main() {
    let client = GraphQLClient::new(
        "http://localhost:4000/graphql",
        reqwest::blocking::Client::new(),
    );
    let result = introspect::run(
        GraphIntrospectInput {
            headers: HashMap::default(),
        },
        &client,
        true,
    );
    match result {
        Ok(response) => println!("{}", response.schema_sdl),
        Err(error) => eprintln!("Unable to introspect:\n{error}"),
    }
}

```
